### ISSUE TITLE
npm install failed on Mac os 10.13.4

### ISSUE THREAD

**Issue by jerry-sl** [LiskHQ/lisk#2082]
### Expected behavior
npm install will install all dependency correctly.
### Actual behavior
npm install failed with errors:
```
checking whether the C compiler works... no
configure: error: in `/Users/myname/github/lisk/lisk/node_modules/sodium/deps/libsodium':
configure: error: C compiler cannot create executables
See `config.log' for more details
make: *** [libsodium] Error 77
/Users/eth/github/lisk/lisk/node_modules/sodium/install.js:288
            throw new Error(cmdLine + ' exited with code ' + code);
            ^

Error: make libsodium exited with code 2
    at ChildProcess.<anonymous> (/Users/eth/github/lisk/lisk/node_modules/sodium/install.js:288:19)
    at ChildProcess.emit (events.js:182:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:237:12)
npm WARN ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! sodium@2.0.1 preinstall: `node install.js --preinstall`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sodium@2.0.1 preinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/myname/.npm/_logs/2018-06-01T09_14_25_225Z-debug.log
```
### Steps to reproduce
pulled 'development' branch at commit: 76b0265d8ce4f11f544779f145855953aa9a05da 
cd ${SRC_DIR};
npm install

### Which version(s) does this affect? (Environment, OS, etc...)
OS Version:       macOS High Sierra
npm version:    5.6.0
node version:   v10.2.1
libtool -V:          Apple Inc. version cctools-906
autoconf version:    autoconf (GNU Autoconf) 2.69

**Comment by lhfly5201314** [comment]
I met the same issue on the Ubuntu!
Any help????

**Comment by Nazgolze** [comment]
"checking whether the C compiler works... no" - sounds like a problem in your setup. Which C compiler are you using?

**Comment by lhfly5201314** [comment]
Hi,friend! I use the gcc version 5.5.0.
Is that ok?

**Comment by lhfly5201314** [comment]
@Nazgolze I upgraded my cmake a few days ago. Does that affect?

**Comment by MaciejBaj** [comment]
@lhfly5201314 did you install necessary system dependencies? 

```
Ubuntu/Debian:

sudo apt-get update
sudo apt-get install -y python build-essential curl automake autoconf libtool


MacOS 10.12-10.13 (Sierra/High Sierra):

Make sure that you have both XCode and Homebrew installed on your machine.

Update homebrew and install dependencies:

brew update
brew doctor
brew install curl automake autoconf libtool
```

**Comment by Slamper** [comment]
This will be fixed by #2140 

The usage of a new libsodium wrapper will resolve all platform build issues of `sodium`

**Comment by diego-G** [comment]
@jerry-sl could you confirm this was solved after #2140 has been merged?

**Comment by jerry-sl** [comment]
@diego-G  solved at 1.1.0 branch, thx.
